### ISSUE TITLE
Anytree and Recursion

### ISSUE THREAD

**Issue by FedeCuci** [c0fec0de/anytree#145]
I am trying to create a script using anytree with the same functionality as this one: https://pastebin.com/tyuPP78W. Basically, this script breaks down how the merge sort algorithm works by nesting output on the terminal based on the level of recursion of where the current code is being run, ultimately making a tree-like diagram. This really helps get a deeper understanding of how recursion works. However, right now it looks kind of ugly and so I want to use anytree to nest output based on the level of recursion of a certain piece of code. However, even after having looked at the documentation and testing a few things, I was not really able to figure out how I could achieve this. 

In other words, how can I set the parent of a node equal to the node from the previous level or recursion? Is this even possible?

I don't know if this is the right place to ask, if so let me know a place where it would be better suited!

**Comment by c0fec0de** [comment]
A tree has no recursion. All parent nodes of a node must not be the node itself.
